snuba/datasets/processors/spans_processor.py

```diff
@@ -0,0 +1,493 @@
+import copy
+import logging
+import numbers
+import random
+import uuid
+from datetime import datetime
+from typing import Any, Dict, Mapping, MutableMapping, MutableSequence, Optional, Tuple
+
+import sqlparse
+from sentry_relay.consts import SPAN_STATUS_NAME_TO_CODE
+
+from snuba import environment, state
+from snuba.consumers.types import KafkaMessageMetadata
+from snuba.datasets.events_format import (
+    EventTooOld,
+    enforce_retention,
+    extract_extra_tags,
+    extract_nested,
+)
+from snuba.datasets.processors import DatasetMessageProcessor
+from snuba.processor import (
+    InsertBatch,
+    ProcessedMessage,
+    _as_dict_safe,
+    _ensure_valid_date,
+    _unicodify,
+)
+from snuba.utils.metrics.wrapper import MetricsWrapper
+
+logger = logging.getLogger(__name__)
+
+metrics = MetricsWrapper(environment.metrics, "spans.processor")
+
+UNKNOWN_SPAN_STATUS = 2
+
+EventDict = MutableMapping[str, Any]
+SpanDict = MutableMapping[str, Any]
+CommonSpanDict = MutableMapping[str, Any]
+RetentionDays = int
+
+
+def is_project_in_allowlist(project_id: int) -> bool:
```

Comment:
Config option to allow spans from a fixed set of projects to be written to the table